mmpose/datasets/datasets/mesh/mesh_mix_dataset.py

```diff
@@ -0,0 +1,40 @@
+from abc import ABCMeta
+
+import numpy as np
+from torch.utils.data import Dataset
+
+from mmpose.datasets.builder import DATASETS
+from .mesh_base_dataset import MeshBaseDataset
+
+
+@DATASETS.register_module()
+class MeshMixDataset(Dataset, metaclass=ABCMeta):
+    """Mix Dataset for 3D human mesh estimation.
+
+    The dataset combines data from multiple datasets (MeshBaseDataset) and
+    sample the data from different datasets with the provided proportions.
+    The dataset loads raw features and apply specified transforms
+    to return a dict containing the image tensors and other information.
+
+    Args:
+        configs (list): List of configs for multiple datasets.
+        partition (list): Sample proportion of multiple datasets.
+    """
+
+    def __init__(self, configs, partition):
+        """Load data from multiple datasets."""
+        self.datasets = [MeshBaseDataset(**cfg) for cfg in configs]
+        self.partition = np.array(partition).cumsum()
+        self.length = max([len(ds) for ds in self.datasets])
```

Comment:
python supports `max(len(ds) for ds in self.datasets)`

Reply:
actually if you want to make samples in one sub-dataset are taken with equal probability, the length should not be got by `max`, but by `prod`.

say two datasets with length 2 and 3, and probabilities are 0.5 and 0.5. 
the max length is 3.
the first sample in the first dataset will be sampled more often than the second sample in the first dataset.

Reply:
however, the length would be too large if we use prod

Reply:
Oh, that's right! Is there other solutions of this problem?

Reply:
what length is proper for the mixed dataset?

Reply:
It should not be too large. How about use the following code:
```
p = np.random.rand()
for i in range(len(self.datasets)):
    if p <= self.partition[i]:
        index_new = (index + np.random.rand()) * len(self.datasets[i]) / self.length
        index_new = np.round(index_new) % (len(self.datasets[i]))
        return self.datasets[i][index_new]
```

Reply:
looks better

Reply:
`int(np.round(index_new)) % ...`
not
`int(np.round(index_new) % ...)`

Reply:
I get it.